src/Clapi/Tree.hs

```diff
@@ -77,7 +77,9 @@ treeApplyReorderings contOps (RtContainer children) =
   let
     attMap = fst <$> contOps
     childMap = Map.foldlWithKey'
-        (\acc k att -> Map.insert k (att, RtEmpty) acc)
+        (\acc k att -> if Map.notMember k acc
```

Comment:
I think you want `Map.alter (Just . maybe (att, RtEmpty) id) k acc`, which would avoid having to lookup stuff in the map twice.